Detectors/TRD/workflow/src/TRDTrapSimulatorSpec.cxx

```diff
@@ -221,25 +223,30 @@ void TRDDPLTrapSimulatorTask::run(o2::framework::ProcessingContext& pc)
 
   // input
   auto inputDigits = pc.inputs().get<gsl::span<o2::trd::Digit>>("digitinput");                                 // block of TRD digits
-  auto digitMCLabels = pc.inputs().get<o2::dataformats::ConstMCTruthContainer<o2::MCCompLabel>>("labelinput"); // MC labels associated to the input digits
   auto inputTriggerRecords = pc.inputs().get<gsl::span<o2::trd::TriggerRecord>>("triggerrecords");             // time and number of digits for each collision
   if (inputDigits.size() == 0 || inputTriggerRecords.size() == 0) {
     LOG(warn) << "Did not receive any digits, trigger records, or neither one nor the other. Aborting.";
     return;
   }
   LOG(debug) << "Read in " << inputDigits.size() << " digits";
-  LOG(debug) << "Labels contain " << digitMCLabels.getNElements() << " elements with and indexed size of " << digitMCLabels.getIndexedSize();
-  LOG(debug) << "Trigger records are available for " << inputTriggerRecords.size() << " collisions";
-  if (digitMCLabels.getIndexedSize() != inputDigits.size()) {
-    LOG(warn) << "Digits and Labels coming into TrapSimulator are of differing sizes, labels will be jibberish. " << digitMCLabels.getIndexedSize() << "!=" << inputDigits.size();
+
+  const o2::dataformats::ConstMCTruthContainer<o2::MCCompLabel>* lblDigitsPtr = nullptr;
+
+  if (mUseMC) {
+    auto lblDigits = pc.inputs().get<o2::dataformats::ConstMCTruthContainer<o2::MCCompLabel>>("labelinput"); // MC labels associated to the input digits
+    lblDigitsPtr = &lblDigits;
+    LOG(debug) << "Labels contain " << lblDigitsPtr->getNElements() << " elements with and indexed size of " << lblDigitsPtr->getIndexedSize();
+    if (lblDigitsPtr->getIndexedSize() != inputDigits.size()) {
+      LOG(warn) << "Digits and Labels coming into TrapSimulator are of differing sizes, labels will be jibberish. " << lblDigitsPtr->getIndexedSize() << "!=" << inputDigits.size();
+    }
   }
+  LOG(debug) << "Trigger records are available for " << inputTriggerRecords.size() << " collisions";
 
   // output
   std::vector<Tracklet64> trapTrackletsAccum;                          // calculated tracklets
-  o2::dataformats::MCTruthContainer<o2::MCCompLabel> trackletMCLabels; // MC labels for the tracklets, taken from the digits which make up the tracklet (duplicates are removed)
   // copy from the input to keep the collision times, but the number of objects in here will refer to tracklets instead of digits
   std::vector<o2::trd::TriggerRecord> trackletTriggerRecords(inputTriggerRecords.begin(), inputTriggerRecords.end()); // time and number of tracklets for each collision
-  trapTrackletsAccum.reserve(inputDigits.size() / 500);
+  o2::dataformats::MCTruthContainer<o2::MCCompLabel> lblTracklets;                                                    // MC labels for the tracklets, taken from the digits which make up the tracklet (duplicates are removed)
```

Comment:
OK, I see, you create the container in any case, so with `--disable-mc` it just stays empty and is not shipped, should be ok then.